qiskit_nature/operators/second_quantization/qubit_converter.py

```diff
@@ -40,58 +106,235 @@ def z2symmetry_reduction(self) -> Optional[List[int]]:
+        return copy.deepcopy(self._z2symmetries)
+
+    def convert(self, second_q_ops: Union[SecondQuantizedOp, List[SecondQuantizedOp]],
+                num_particles: Optional[Tuple[int, int]] = None,
+                z2symmetry_reduction: Optional[List[int]] = None,
+                ) -> Union[PauliSumOp, List[Optional[PauliSumOp]]]:
         """
         Maps the given list of second quantized operators to qubit operators. Also it will
         carry out z2 symmetry reduction on the qubit operators if z2symmetry_reduction has
-        been specified.
+        been specified. For convenience a single operator may be passed, without wrapping
+        it in a List, and in which case a single qubit operator will likewise be returned.
 
         Args:
-            second_q_ops: The list of second quantized operators to be converted
+            second_q_ops: A second quantized operator, or list thereof to be converted.
+            num_particles: Needed for two qubit reduction to determine correct sector. If
+                not supplied, even if two_qubit_reduction is possible, it will not be done.
             z2symmetry_reduction: Optional z2symmetry reduction, the sector of the symmetry
 
         Returns:
-            A list of qubit operators or the same length as the second_q_ops list. The first
+            A qubit operator or list of thereof the same length as the second_q_ops list. The first
             operator in the second_q_ops list is treated as the main operator and others must
             commute with its symmetry, when symmetry reduction is being done. If it does not
             then the position in the output list will be set to `None` to preserve the order.
         """
+        self._invalidate()  # Invalidate state before conversion is attempted
+
+        # To allow a single operator to be converted, but use the same logic that does the
+        # actual conversions, we make a single entry list of it here and unwrap to return.
+        wrapped = False
+        if isinstance(second_q_ops, SecondQuantizedOp):
+            second_q_ops = [second_q_ops]
+            wrapped = True
+
         if z2symmetry_reduction is not None:
             self.z2symmetry_reduction = z2symmetry_reduction
 
         qubit_ops = self._map_to_qubits(second_q_ops)
-        qubit_ops_reduced = self._symmetry_reduce(qubit_ops)
+        qubit_ops_reduced, did_2_q = self._two_qubit_reduce(qubit_ops, num_particles)
+        qubit_ops_tapered, z2symmetries = self._find_symmetry_reduce(qubit_ops_reduced)
+
+        # Set the state (i.e. whether we did two qubit reductions and the Z2Symmetries)
+        # now that a conversion was successful
+        self._set_valid(did_2_q, num_particles, z2symmetries)
 
-        return qubit_ops_reduced
+        if wrapped:
+            qubit_ops_tapered = qubit_ops_tapered[0]
+
+        return qubit_ops_tapered
+
+    def convert_more(self, second_q_ops: Union[SecondQuantizedOp, List[SecondQuantizedOp]]
```

Comment:
I am wondering whether it may be nice to merge this and the previous method (at least for the public API) into a single `convert` method. `num_particles` and `z2symmetry_reductions` are `Optional` already and we can leverage that to our advantage here.

For example:
```python
converter = QubitConverter(mapper=ParityMapper(), two_qubit_reduction=True)

initial_qubit_op = converter.convert(initial_second_q_op, num_particles)

# At this point, `_check_valid` will return True and we already have all the information necessary. Omitting the optional args can thus trigger the current `convert_more` function.
more_qubit_ops = converter.convert(more_second_q_ops)
```

If a user would want to re-use a converter object without using the state (for which I cannot think of a reason but who knows...) he can simply invalidate the state before calling `convert` again.

Reply:
I only just added the convert_more. They are different but similar. I am more trying to get the code together and tested at present. If you want two q reduction and auto symmetry reduction then the user has to supply num particles. When converting again it uses the stored num particles from before.

Reply:
Yes, that's what I am saying. I don't think it's necessary to have this in two methods because the `convert` method itself could reuse the stored values (if they are available).

Reply:
My feeling at present is that the convertor should just convert, given the set of parameters it has just been passed, without relying on magic values in there. Sector can be passed on constructor so user can set things manually, though given something like auto sector that can be overridden. So already it has value tests. My preference is that if you want to convert the same was as a prior one, and supply just 2nd q ops, that you do it explicitly. This way there should be less issues of having values that unexpectedly cause the convert to use any stored state it has.

Reply:
Hm I see. But that would require the code to be changed such that you can pass in `Z2Symmetries` rather than a sector right? Where would these `Z2Symmetries` be stored? Would the initial `convert` have to return them too?

Reply:
No it needs a sector - the sector can be put as tapering_values in Z2Symmetry once symmetries are found, to tell the taper method to just give back the operator for that sector not all of them,